tests/api/printf.cpp

```diff
@@ -104,6 +106,36 @@ static std::string getStdoutFileName(temp_folder_deletion& temp) {
     return (prefix / suffix).string();
 }
 
+void printf_callback(const char* buffer, size_t len, size_t complete,
+                     void* user_data) {
+    auto org_buf_size = globalBuffer.size();
+    // Calculate how much data we can fit
+    size_t space_available = globalBuffer.capacity() - org_buf_size;
+    size_t data_to_copy = (len <= space_available) ? len : space_available;
+    // Copy data into the buffer (up to the available space)
+    if (space_available > len) {
+        globalBuffer.resize(org_buf_size + len, '\0');
+        memcpy(globalBuffer.data() + org_buf_size, buffer, data_to_copy);
+    }
+
+    if (complete || (space_available < len && space_available >= 1)) {
+        globalBuffer.emplace_back('\0');
+    }
+}
+
+std::vector<cl_context_properties>& setup_arm_printf_test(long int buff_size) {
+    globalBuffer.clear();
+    globalBuffer.shrink_to_fit();
+    globalBuffer.reserve(buff_size);
```

Comment:
Why do you want to reserve `buff_size`? This is just complicated things I feel. `globalBuffer` has nothing to do with the `cl_arm_printf` extension, it does not have to use a specific size reserve.